plugin/trino-open-policy-agent/src/main/java/io/trino/plugin/openpolicyagent/OpaAccessControlFactory.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.openpolicyagent;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import io.airlift.bootstrap.Bootstrap;
+import io.airlift.http.client.HttpClient;
+import io.airlift.json.JsonModule;
+import io.trino.spi.security.SystemAccessControl;
+import io.trino.spi.security.SystemAccessControlFactory;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static io.airlift.http.client.HttpClientBinder.httpClientBinder;
+import static io.airlift.json.JsonCodecBinder.jsonCodecBinder;
+import static java.util.Objects.requireNonNull;
+
+public class OpaAccessControlFactory
+        implements SystemAccessControlFactory
+{
+    @Override
+    public String getName()
+    {
+        return "opa";
+    }
+
+    @Override
+    public SystemAccessControl create(Map<String, String> config)
+    {
+        return create(config, Optional.empty());
+    }
+
+    @VisibleForTesting
+    protected SystemAccessControl create(Map<String, String> config, Optional<HttpClient> httpClient)
+    {
+        requireNonNull(config, "config is null");
+
+        Bootstrap app = new Bootstrap(
+                new JsonModule(),
+                binder -> {
+                    jsonCodecBinder(binder).bindJsonCodec(OpaQuery.class);
+                    jsonCodecBinder(binder).bindJsonCodec(OpaQueryResult.class);
+                    if (httpClient.isEmpty()) {
+                        httpClientBinder(binder).bindHttpClient("opa-access-control", ForOpa.class);
+                    }
+                    else {
+                        binder.bind(Key.get(HttpClient.class, ForOpa.class)).toInstance(httpClient.orElseThrow());
+                    }
```

Comment:
Considering this is only used for testing I thought this would be a reasonable compromise. Happy to change it if you think it's worth the extra complexity though

Reply:
I prefer to keep testing code as much separated as possible. I know this is not always feasible, though. So I won't be pushing for this, but if you think of some simpler way to make the separation, that would be nice :)

Reply:
Agreed. All I can think of (from quickly glancing at a few other plugins) is to either:
- Build the authorizer using the constructor instead of relying on the factory and injection (e.g. what the Pinot module does), or
- Spin up a real test HTTP server instead of using a mock HTTP client (e.g. what the HTTP event listener module does)

The former would work, but given the number of objects that we're building through injection now (the `OpaHttpClient`, `OpaHighLevelClient`, etc...) I feel it could get messy. The latter would work but it'd be a larger scale refactor just to remove this line of code.

I'll resolve for now, but if anyone has objections or nicer ideas please reopen :)